src/engraving/libmscore/note.cpp

```diff
@@ -1436,6 +1436,13 @@ void Note::draw(mu::draw::Painter* painter) const
                 painter->setPen(selected() ? engravingConfiguration()->warningSelectedColor() : engravingConfiguration()->warningColor());
             }
         }
+        // Warn if notes are unplayable based on previous harp diagram setting
```

Comment:
Every notehead causes a search for a previous diagram currently - I'm thinking this isn't sustainable and a change of approach might be necessary

Reply:
Many similar things (clef, key sig, time sig) are stored in the Staff class, indeed as a `tick`-to-`…` map. Maybe that'd work for harp pedal state too?

(Disadvantage is of course that it would require extra effort to apply it to both staves instead of only the staff to which the diagram belongs technically. However, I believe this is a more general problem, that applies to dynamics too.)

Reply:
Yes, I think it'd sort a few inefficiencies.  Would storing it in the `Part` class be an acceptable solution to make it appear on both staves?

Reply:
Interesting idea. It looks like the Part class has not really been used yet for storing data like this, so I'm not completely sure what the implications might be, but we could definitely try it. Maybe it turns out to be a very good idea.

Reply:
It definitely does what I want but I'm not sure what the wider implications would be!  There don't seem to be any issues so far.